Signal/src/util/DateUtil.m

```diff
@@ -47,12 +47,46 @@ + (NSDateFormatter *)timeFormatter {
     return formatter;
 }
 
++ (NSDateFormatter *)monthAndDayFormatter
+{
+    static NSDateFormatter *formatter;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        formatter = [NSDateFormatter new];
+        [formatter setLocale:[NSLocale currentLocale]];
+        formatter.dateFormat = @"MMM d";
+    });
+    return formatter;
+}
+
++ (NSDateFormatter *)shortDayOfWeekFormatter
+{
+    static NSDateFormatter *formatter;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        formatter = [NSDateFormatter new];
+        [formatter setLocale:[NSLocale currentLocale]];
+        formatter.dateFormat = @"E";
+    });
+    return formatter;
+}
+
 + (BOOL)dateIsOlderThanOneDay:(NSDate *)date {
-    return [[NSDate date] timeIntervalSinceDate:date] > kDayInterval;
+    NSDate *now = [NSDate date];
+    NSCalendar *calendar = [NSCalendar currentCalendar];
+
+    NSUInteger dateDayOfEra = [calendar ordinalityOfUnit:NSCalendarUnitDay inUnit:NSCalendarUnitEra forDate:date];
+    NSUInteger nowDayOfEra = [calendar ordinalityOfUnit:NSCalendarUnitDay inUnit:NSCalendarUnitEra forDate:now];
+    return dateDayOfEra < nowDayOfEra;
 }
 
 + (BOOL)dateIsOlderThanOneWeek:(NSDate *)date {
-    return [[NSDate date] timeIntervalSinceDate:date] > kWeekInterval;
```

Comment:
Similarly, the intent here isn't to measure a fixed amount of seconds, but to say "more than 6/7 (depending on how you count) days have passed".

Reply:
:+1: